packages/components/src/internal/query/api.ts

```diff
@@ -994,6 +994,39 @@ export function saveRows(options: Query.SaveRowsOptions): Promise<Query.SaveRows
     });
 }
 
+function splitRowsByContainer(rows: any[], containerField: string): Record<string, any[]> {
+    const containerRows = {};
+    rows.forEach(row => {
+        const container = caseInsensitive(row, containerField);
+        if (!containerRows[container]) containerRows[container] = [];
+        containerRows[container].push(row);
+    });
+
+    return containerRows;
+}
+
+export function saveRowsByContainer(options: Query.SaveRowsOptions, containerField: string = 'ContainerPath'): void {
+    const commands = [];
+
+    // for each original command, split it into multiple commands for each container in the rows
+    options.commands.forEach(command => {
```

Comment:
nit: Seems a bit odd that the `containerField` defaults to `"ContainerPath"` which is not a normal field in LKS. I would expect maybe `Folder/Path`?